### PR TITLE
[site] Add links to new documentation from the site

### PR DESCRIPTION
**Pull request by jwnrt**
## Summary

This PR has mostly the same intent as https://github.com/lowRISC/opentitan/pull/17400 but splits the CSS into separate files. It also includes a fix for the tiles on the documentation page in darkmode.

I made this as a separate PR to #17400 to preserve the original commits we made when working on this.

This PR adds links into the mdbook from the site.

* Links in the header and footer to documentation, getting started, and dashboard.
* New "documentation" page with links into different parts of the documentation.

This work was done separately elsewhere and factored into these commits.

There is some refactoring happening here to make the new page easier to add:

* Style for the landing page moved to `landing.css`
* Style and layout for the info pages moved to `info.html` and `info.css`
* Header, footer, and theme selector extracted out of `index.html`.

This allows us to add a new page layout (`single.html` and `single.css`) for pages like the documentation page.

## Screenshots

This screenshot shows the new header, footer, and the documentation page. This header and footer is now also used on the landing page.

image

### CODE REVIEW

**Comment by hcallahan-lowrisc** [comment]
@jwnrt How happy are you with the changes here as it stands? I'm inclined to close #17400 in favor of this, but I am more familiar with the changes there just because I've been using them on staging.opentitan.org. 
Shall I close that PR and ask people to review this one? Or should I get that one merged and then rebase this on top of master as a follow up?

**Comment by jwnrt** [comment]
> @jwnrt How happy are you with the changes here as it stands?

I'm happy with these changes, don't think there's anything else I wanted to do here.

> I'm inclined to close #17400 in favor of this, but I am more familiar with the changes there just because I've been using them on staging.opentitan.org. Shall I close that PR and ask people to review this one? Or should I get that one merged and then rebase this on top of master as a follow up?

Since these changes are the same as #17400, I'm happy for this PR to be reviewed instead. It has the same effect, just with the original commit history preserved from when we first did this work, plus some other little fixes and tidying up.

**Comment by jwnrt** [comment]
Rebased

**Review by hcallahan-lowrisc — COMMENTED** [review]
Can we change the links in the headers/footers/boxes to be relative? 
e.g. https://github.com/lowRISC/opentitan/commit/36e4187123b2537b59a0656089410924f73ec55e
This lets me easily host the site at a path e.g. `staging.opentitan.org/pr/<pr_number>` etc.

File: site/landing/content/trademark-usage-policy.md [hunk 1]
```diff
@@ -1,13 +1,14 @@
 ---
 title: "OpenTitan Trademark Usage Policy"
+layout: "info"
```
Comment (engdoreis) [inline site/landing/content/trademark-usage-policy.md:3]: Does mdbook support this tags? I thought it was Hugo tags.

Comment (jwnrt) [reply site/landing/content/trademark-usage-policy.md:3]: The main website (`site/landing/`) is still a Hugo site, so we can still use these tags

File: site/landing/layouts/index.html [hunk 2]
```diff
@@ -293,65 +258,15 @@ <h2 class="sr-only">Join the Project</h2>
...
-    <p>No license is granted for the OpenTitan logo or other trademarks. <a
-      href="/trademark-usage-policy/">Trademark Usage Policy</a>.</p>
-  </footer>
+  {{ partial "footer.html" . }}
 
   <script src="/js/tiny-slider.js"></script>
+  <script src="/js/theme-changer.js"></script>
 
   <script>
+	// Enable JS features
```
Comment (HU90m) [inline site/landing/layouts/index.html:267]: nit: This is tabbed, the rest uses spaces.

**Review by HU90m — APPROVED** [review]
Looks good. Only one nit.

File: site/landing/layouts/index.html (hunk 2 above)
Comment (jwnrt) [reply site/landing/layouts/index.html]: My bad, I only recently made my editor change tabs to spaces.

**Review by HU90m — APPROVED** [review]
Thanks for fixing my nit :smile:

**Comment by jwnrt** [comment]
Squashed fixup

**Review by engdoreis — APPROVED** [review]
I didn't review the HTML and CSS, but the .md files LGTM.

**Review by marnovandermaas — APPROVED** [review]
Thanks for these changes. I didn't render them, but the code changes look good.

**Pull request merged**